datafusion/sql/src/unparser/expr.rs

```diff
@@ -859,6 +878,35 @@ impl Unparser<'_> {
         }
     }
 
+    fn process_interval_nanosecond(nano: i64) -> String {
```

Comment:
There seems to be similar code already in https://github.com/apache/arrow-rs/blob/72467c670f8c38130e4743347407f1a542e59e0c/arrow-cast/src/display.rs#L600-L649

I think we could potentially use it via

https://docs.rs/arrow/latest/arrow/util/display/fn.array_value_to_string.html

After first converting ScalarValue to a one value array: https://docs.rs/datafusion/latest/datafusion/common/enum.ScalarValue.html#method.to_array

Reply:
Thanks @alamb. It looks great. I'll address this today. :)

Reply:
@alamb After using ArrowFormatter, I noticed that the results contain many redundant words. For example:

```
input expr: interval_month_day_nano_lit("-3 MONTH"),
output result: r#"INTERVAL '0 YEARS -3 MONS 0 DAYS 0 HOURS 0 MINS 0.000000000 SECS'"#,
---
input expr: interval_month_day_nano_lit("1 YEAR 1 MONTH 1 DAY 3 HOUR 10 MINUTE 20 SECOND"),
output result: r#"INTERVAL '0 YEARS 13 MONS 1 DAYS 3 HOURS 10 MINS 20.000000000 SECS'"#,
```
I think both of these are valid SQL. They just don't look very smart, but it's okay with me. 

I implemented the code like
```rust
            ScalarValue::IntervalMonthDayNano(Some(_i)) => {
                let wrap_array = v.to_array()?;
                let Some(result) = array_value_to_string(&wrap_array, 0).ok() else {
                    return internal_err!("Unable to convert IntervalMonthDayNano to string");
                };
                let interval = Interval {
                    value: Box::new(ast::Expr::Value(SingleQuotedString(result.to_uppercase()))),
                    leading_field: None,
                    leading_precision: None,
                    last_field: None,
                    fractional_seconds_precision: None,
                };
                Ok(ast::Expr::Interval(interval))
            }
```
What do you think? Does it make sense?

Reply:
hmm.. OK.  I think it would be an enhancement for the arrow formatter. If we want to make it pretty, I think we should modify the behavior in `arrow-cast` crate.

Reply:
> Hmm... OK. I think it would be an enhancement for the ArrowFormatter. If we want to make it smarter, I think we should modify the behavior in the `arrow-cast` crate.

I agree -- I filed https://github.com/apache/arrow-rs/issues/5914 to track this suggestion